qml/Stage/Stage.qml

```diff
@@ -962,7 +962,11 @@ FocusScope {
                         priv.updateMainAndSideStageIndexes();
                     }
                     appDelegate.focus = true;
-                    priv.focusedAppDelegate = appDelegate;
+
+                    // Don't set focusedAppDelegate (and signal mainAppChanged) unnecessarily
+                    // which can happen after getting interactive again.
+                    if (priv.focusedAppDelegate !== appDelegate)
+                        priv.focusedAppDelegate = appDelegate;
```

Comment:
Are you sure this is needed? I think that the QML JS engine does not emit the signal if you assign the same value to a variable.

Reply:
Apparently not. Otherwise, the trick [like the one in camera-app](https://gitlab.com/ubports/apps/camera-app/blob/master/ViewFinderOverlay.qml#L244) won't work.

Reply:
Well, I guess we have a couple of things to fix, then:-)
Please run this small file with qmlscene :-)
```
import QtQuick 2.2
import QtQuick.Window 2.2

Window {
    width: 400
    height: 300
    visible: true

    QtObject {
        id: priv
        property string name: "tommys"
        onNameChanged: console.log("Name changed!")
    }
    Timer {
        repeat: true
        interval: 100
        onTriggered: priv.name = "tommy"
        running: true
    }
}
```
You'll see only one name changed signal being generated.

Reply:
That might applies to `property string` only?

```qml
import QtQuick 2.2
import QtQuick.Window 2.2

Window {
    width: 400
    height: 300
    visible: true

    QtObject {
        id: priv
        property string name: "tommys"
        onNameChanged: console.log("Name changed!")

        property var name2: "name2"
        onName2Changed: console.log("Name2 changed!")

        property var obj: {}
        onObjChanged: console.log("obj changed!")
    }

    Timer {
        repeat: true
        interval: 1000
        onTriggered: {
            priv.name = priv.name
            priv.name2 = priv.name2
            priv.obj = priv.obj
        }
        running: true
    }
}
```

`nameChanged` is not signaled, but `name2Changed` and `objChanged` is. Tested on Qt 5.9.5 (in Ubuntu 18.04) and Qt 5.14.0 (official Qt build).

Reply:
Ouch, that's kind of weird :-) Anyway, then I have no complaints!